include/seastar/core/deleter.hh

```diff
@@ -113,7 +114,7 @@ private:
 
 /// \cond internal
 struct deleter::impl {
-    unsigned refs = 1;
+    std::atomic<unsigned int> refs = 1;
```

Comment:
Include a reference to the JIRA issue, and augment the doc to describe the new behavior and safety contract.